lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -270,17 +272,12 @@ def filter_query(filters: list[str], query: Query) -> Query:
                         HTTPStatus.UNAUTHORIZED,
                         f"Metadata key {k} cannot be used by an unauthenticated client",
                     )
-                user_private = [Metadata.user_id == user_id]
 
             if not use_dataset:
-                expression = Metadata.value[keys].as_string()
-                terms = [
-                    Metadata.key == native_key,
-                    expression.contains(v) if contains else expression == v,
-                ]
-                if user_private:
-                    terms.extend(user_private)
-            filter = and_(*terms)
+                expression = aliases[native_key].value[keys].as_string()
+                filter = expression.contains(v) if contains else expression == v
+            else:
+                filter = and_(*terms)
```

Comment:
I agree, there's nothing critical here.

However, as the code stands with this change, I don't think we need for the two `if` statements (lines 246 and 276 respectively) to be separate from each other (that is, the path through the first dictates the path through the second).

So, I think it would be reasonable (even good!) to combine them together.  We can get rid of the `use_dataset` flag, and the scope of `terms` becomes limited to only one block (and, in fact, becomes so simple that we can get rid of it altogether).  You would have to replace the `elif` with an `else`, but that's easily done by moving the conditional to the contained `if`.  And, the result would be much more straightforward code.

Reply:
Your latest revision is much improved.  However, I still suggest combining the two `if` statements, if you opt to make a change for some other reason.